src/kw/kw.cpp

```diff
@@ -372,16 +383,72 @@ bool KanoWorld::save_data()
     json_object_set_string(
         json_object(user_data.get()), "token", get_token().c_str());
     json_object_set_string(
-        json_object(user_data.get()), "duration", get_expiration_date().c_str());
+        json_object(user_data.get()), "duration",
+                    get_expiration_date().c_str());
+    json_object_set_boolean(
+        json_object(user_data.get()), "is_verified",
+                    this->is_verified_cache);
 
-    JSON_Status rc = json_serialize_to_file(user_data.get(), data_filename.c_str());
+    JSON_Status rc = json_serialize_to_file(
+        user_data.get(), data_filename.c_str());
 
     return rc == JSONSuccess;
 }
 
+
 /**
- * \warning Not implemented
+ * \warning Implementation not finalised
  */
-bool KanoWorld::is_account_verified() {
-    return true;
+bool KanoWorld::is_account_verified(const bool cache) {
+    this->load_data();
+
+    if (cache && this->is_verified_cache) {
+        return true;
+    }
+
+    this->is_verified_cache = this->is_account_verified_api();
+    this->save_data();
+
+    return this->is_verified_cache;
+}
+
+
+void KanoWorld::clear_account_verified_cache() {
+    this->load_data();
+    this->is_verified_cache = false;
+    this->save_data();
+}
+
+
+/**
+ * \warning Implementation not finalised
+ */
+bool KanoWorld::is_account_verified_api() {
```

Comment:
Seems like this might be a candidate for:

```suggestion
bool KanoWorld::is_account_verified_api() const {
```